plugins/parsers/csv/parser_test.go

```diff
@@ -0,0 +1,193 @@
+07/11/09 04:05:06 PM,80,test_name2`
+	metrics, err := p.Parse([]byte(testCSV))
+	require.NoError(t, err)
+	require.NotEqual(t, metrics[1].Time(), metrics[0].Time())
+}
+
+func TestTimestampError(t *testing.T) {
+	p := CSVParser{
+		Header:          true,
+		DataColumns:     []string{"first", "second", "third"},
+		FieldColumns:    []string{"second"},
+		NameColumn:      "third",
+		TimestampColumn: "first",
+	}
+	testCSV := `line1,line2,line3
+23/05/09 04:05:06 PM,70,test_name
+07/11/09 04:05:06 PM,80,test_name2`
+	_, err := p.Parse([]byte(testCSV))
+	require.Equal(t, fmt.Errorf("timestamp format must be specified"), err)
+}
+
+func TestQuotedCharacter(t *testing.T) {
+	p := CSVParser{
+		Header:       true,
+		DataColumns:  []string{"first", "second", "third"},
+		FieldColumns: []string{"second", "first"},
+		NameColumn:   "third",
+	}
+
+	testCSV := `line1,line2,line3
+"3,4",70,test_name`
+	metrics, err := p.Parse([]byte(testCSV))
+	require.NoError(t, err)
+	require.Equal(t, "3,4", metrics[0].Fields()["first"])
+}
+
+func TestDelimiter(t *testing.T) {
+	p := CSVParser{
+		Header:       true,
+		Delimiter:    "%",
+		DataColumns:  []string{"first", "second", "third"},
+		FieldColumns: []string{"second", "first"},
+		NameColumn:   "third",
+	}
+
+	testCSV := `line1%line2%line3
+3,4%70%test_name`
+	metrics, err := p.Parse([]byte(testCSV))
+	require.NoError(t, err)
+	require.Equal(t, "3,4", metrics[0].Fields()["first"])
+}
+
+func TestValueConversion(t *testing.T) {
+	p := CSVParser{
+		Header:       false,
+		Delimiter:    ",",
+		DataColumns:  []string{"first", "second", "third", "fourth"},
+		FieldColumns: []string{"second", "first", "third", "fourth"},
+		MetricName:   "test_value",
+	}
+	testCSV := `3.3,4,true,hello`
+
+	expectedFields := map[string]interface{}{
+		"first":  3.3,
+		"second": 4,
+		"third":  true,
+		"fourth": "hello",
+	}
+
+	metrics, err := p.Parse([]byte(testCSV))
+	require.NoError(t, err)
+
+	expectedTags := make(map[string]string)
+	goodMetric, err1 := metric.New("test_value", expectedTags, expectedFields, time.Unix(0, 0))
+	returnedMetric, err2 := metric.New(metrics[0].Name(), metrics[0].Tags(), metrics[0].Fields(), time.Unix(0, 0))
+	require.NoError(t, err1)
+	require.NoError(t, err2)
+
+	//deep equal fields
+	require.True(t, reflect.DeepEqual(goodMetric.Fields(), returnedMetric.Fields()))
+}
+
+func TestSkipComment(t *testing.T) {
+	p := CSVParser{
+		Header:       false,
+		Comment:      "#",
+		DataColumns:  []string{"first", "second", "third", "fourth"},
+		FieldColumns: []string{"second", "first", "third", "fourth"},
+		MetricName:   "test_value",
+	}
+	testCSV := `#3.3,4,true,hello
+4,9.9,true,name_this`
+
+	expectedFields := map[string]interface{}{
+		"first":  int64(4),
+		"second": 9.9,
+		"third":  true,
+		"fourth": "name_this",
+	}
+
+	metrics, err := p.Parse([]byte(testCSV))
+	require.NoError(t, err)
+	require.Equal(t, true, reflect.DeepEqual(expectedFields, metrics[0].Fields()))
```

Comment:
`require.Equal(t, expectedFields, metrics[0].Fields())`

Check other tests and make sure you are using this everywhere.